cantusdb/turtle/csv2rdf_by_RDFLib.py

```diff
@@ -3,64 +3,60 @@
-    ontology_f = open("ontology_dict.csv", 'r')
-    ontology_reader = csv.DictReader(ontology_f)
-    ontology_dict = {}
-    
-    # converting the reader dict to the predicate dictionary
-    for row in ontology_reader:
-        ontology_dict[row[list(row.keys())[0]]] = row[list(row.keys())[1]]
-                
-    return ontology_dict
+    with open("relations_mapping.json", 'r') as mapper:
+        # a specific config file for the predicates
+        # TODO: maybe change this part to make a more customized way of getting the predicates configurations
+        data = json.load(mapper)
+    return data
     
 # a main operation function
 def convert_csv_to_turtle():
-    # TODO: maybe change this one to accomadate for multiple files
     if len(sys.argv) != 2:
-        raise ValueError("Invalid number of input filename") 
-    
-    g = Graph()
-
-    csv_file = open(sys.argv[1], 'r', encoding='utf-8')
-    
-    csv_reader = csv.DictReader(csv_file)
-    ontology_dict = get_ontology_dict()
+            raise ValueError("Invalid number of input filename") 
+        
+    with open(sys.argv[1], 'r', encoding='utf-8') as csv_file:
+        # TODO: maybe change this one to accomadate for multiple files
 
-    # Convert each row to Turtle format and add it to the output
-    for row in csv_reader:
-        # the first column as the subject
-        absolute_uri = URIRef(row[list(row.keys())[0]])
-        g.add((absolute_uri, RDF.type, URIRef("http://www.wikidata.org/prop/direct/chant")))
+        g = Graph()
         
-        # extracting other informations
-        for k in row:
-            # if this key is the first column, skip
-            if k.__eq__(list(row.keys())[0]):
-                continue
-            
-            # finding the predicate from csv in the config dictionary, if not exit, skip
-            if k in list(ontology_dict.keys()):
-                predicate = URIRef(ontology_dict[k])
-            else:
-                continue
-            
-            # the object might be an URI or a literal
-            if validators.url(row[k]):
-                obj = URIRef(row[k])
-            else:
-                obj = Literal(row[k])
-            
-            # add this triple (absolute uri, predicate, object) to the graph
-            g.add((absolute_uri, predicate, obj))    
+        csv_reader = csv.reader(csv_file)
+        ontology_dict = get_ontology_dict()
         
-    csv_file.close()
-    
+        header = next(csv_reader)
+
+        # Convert each row to Turtle format and add it to the output
+        for row in csv_reader:
+            # the first column as the subject
+            key_attribute = URIRef(row[0])
+            if "type" in list(ontology_dict.keys()):
+                g.add((key_attribute, RDF.type, URIRef(ontology_dict["type"])))
+            
+            # extracting other informations
+            # TODO: is the source description texts needed?
+            for i in range(len(row)):
```

Comment:
I think you could simplify this `for` block by: 

1. using the `enumerate` function (this way you can iterate through row and get both the index and the value.
2. iterating through `row[1:]` if you don't want the first column

Reply:
Ok. I will fix it.